pil-analyzer/src/statement_processor.rs

```diff
@@ -647,4 +653,29 @@ where
             functions,
         }
     }
+
+    pub fn process_trait_implementation(
+        &self,
+        trait_impl: parsed::TraitImplementation<parsed::Expression>,
+    ) -> TraitImplementation<Expression> {
+        let type_vars = trait_impl.type_scheme.vars.vars().collect();
+        let functions = trait_impl
+            .functions
+            .iter()
+            .map(|named| NamedExpression {
+                name: named.name.clone(),
+                body: Box::new(
+                    self.expression_processor(&type_vars)
+                        .process_expression(named.body.as_ref().clone()),
+                ),
+            })
+            .collect();
+
+        TraitImplementation {
+            name: self.driver.resolve_decl(&trait_impl.name),
```

Comment:
I think this should use `resolve_ref` with TraitDeclaration as symbol category.

Reply:
Also maybe we could rename `name` to `trait`? But not sure if that makes it easier to understand.